### PR TITLE
Doc(mkdocs): Add section for contribution guide

### PR DESCRIPTION
**Pull request by titom73**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [x] Documentation content changes
- [ ] Other (please describe):

## Component(s) name

MKDOCS

## Proposed changes

__Add following pages:__

- Contribution overview
- Getting-started to contribute
- Code style

__Todo:__

- [ ] Add page for PR tasks list and commands

## How to test

```
make start
```

Open http://localhost:8000

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/docs/contribution/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+## Pull requests
+
+* Be sure to open an issue **before** starting work on a pull request, and
+discuss your idea with the __ansible-avd__ maintainers before beginning work. This will
+help prevent wasting time on something that might we might not be able to
+implement. When suggesting a new feature, also make sure it won't conflict with
+any work that's already in progress.
+
+* Any pull request which does _not_ relate to an accepted issue will be closed.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/getting-started.md]: Should we be a bit more service minded and just say `Any pull request which does _not_ relate to an accepted issue will not be approved`

File: ansible_collections/arista/avd/docs/contribution/overview.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+* Keep in mind that bugs are prioritized based on their severity and how much work is required to resolve them. It may take some time for someone to address your issue.
+
+#### Feature Requests
+
+* First, check the GitHub [issues
+list](https://github.com/aristanetworks/ansible-avd/issues) and [Discussions](https://github.com/aristanetworks/ansible-avd/discussions) to see if the feature you're requesting is already listed. You can greater visibility on the committed by looking at the [Arista Validated Design (AVD) Roadmap](https://www.networktocode.com/Arista Validated Design (AVD)/roadmap) (Be sure to search closed issues as well, since some feature requests have not have been accepted.) If the feature you'd like to see has already been requested and is open, click "add a reaction" in the top right corner of the issue and add a thumbs up (+1). This ensures that the issue has a better chance of receiving attention. Also feel free to add a comment with any additional justification for the feature. (However, note that comments with no substance other than a "+1" will be deleted. Please use GitHub's reactions feature to indicate your support.)
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/overview.md]: Why do we have references to "https://www.networktocode.com" ?

**Review by ClausHolbechArista — COMMENTED** [review]
First review.

File: ansible_collections/arista/avd/docs/contribution/style-guide.md [hunk 3]
```diff
@@ -0,0 +1,263 @@
+# AVD Coding style
+
+This page provides a list of guidelines to apply when developing Jinja2 template in AVD context. These rules apply for either creation or update of any J2 file available in aristanetworks/ansible-avd repository.
+
+## Python code style
+
+As AVD is ansible collection, it is require to follow guidelines from [ansible documentation](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_best_practices.html) for all Python code.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/style-guide.md]: is an Ansible collection

File: ansible_collections/arista/avd/docs/contribution/style-guide.md [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+```jinja
+{% set ip_helper_cli = " source-interface " ~ vlan_interfaces[vlan_interface].ip_helpers[ip_helper].source_interface %}
+```
+
+
+### VAR-6 - Variable type comparison
+
+__Description__
+
+To test type of a variable, it is recommended to use `is`/`is ne` keywords
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/style-guide.md]: `is ne` or `is not` ? `is ne` is only used in the codebase as a string match (which is wrong according to VAR-7). `is not` is used widely.

File: ansible_collections/arista/avd/docs/contribution/style-guide.md [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+__Example__
+
+```jinja
+{# Test if variable is string #}
+{% if ethernet_interface == 'Ethernet1' %}
+
+{# Test if variable is not a string #}
+{% if ethernet_interface != 'Ethernet1' %}
+```
+
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/style-guide.md:200]: Maybe add comment to look at PLUGIN-2

File: ansible_collections/arista/avd/docs/contribution/overview.md (hunk 2 above)
Comment (titom73) [reply ansible_collections/arista/avd/docs/contribution/overview.md]: Nope, Incorrect URL. Will be removed

File: ansible_collections/arista/avd/docs/contribution/getting-started.md (hunk 1 above)
Comment (titom73) [reply ansible_collections/arista/avd/docs/contribution/getting-started.md]: Fixed

File: ansible_collections/arista/avd/docs/contribution/style-guide.md (hunk 4 above)
Comment (titom73) [reply ansible_collections/arista/avd/docs/contribution/style-guide.md]: Fixed

File: ansible_collections/arista/avd/docs/contribution/style-guide.md (hunk 5 above)
Comment (titom73) [reply ansible_collections/arista/avd/docs/contribution/style-guide.md:200]: Link added

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Pull request merged**